### PR TITLE
Bump faraday to 1.10.0

### PR DESCRIPTION
**Pull request by ryan-mcneil**
## Summary

- Bump faraday to 1.10.0
- Step 2 in upgrading faraday
- Previous step was replacing the blocking gem - PR [here](https://github.com/department-of-veterans-affairs/vets-api/pull/14544)

Major notes about changes
- the `body` attribute in `Env` objects was split into a more explicit `request_body` and `response_body` attrs. While `body` is still an alias for these fields and handles most cases, there were some updates that needed to be made, such as updating `body` in the instantiation of `Faraday::Request` or `Faraday::Response` objects
- `Faraday::ClientError` used to cover all 4XX & 5XX errors, but now `Faraday::ClientError` covers 4XX and `Faraday::ServerError` covers 5XX
- In a Faraday client, the associated adapter used to be included in the `handlers` property, but was abstracted to it's own prop, `adapter`

## Related issue(s)
- https://app.zenhub.com/workspaces/reliability-team-633b069d2920b776613c93d8/issues/gh/department-of-veterans-affairs/va.gov-team/69303

## Testing done

- Passing test suite
- Requesting smoke testing from VFS teams prior to prod release

### CODE REVIEW

File: lib/common/client/base.rb [hunk 1]
```diff
@@ -73,8 +73,9 @@ def connection
         @connection ||= lambda do
           connection = config.connection
           handlers = connection.builder.handlers
+          adapter = connection.builder.adapter
```
Comment (ryan-mcneil) [inline lib/common/client/base.rb:76]: the Faraday::Adapter was moved into it's own explicit object in Faraday 1+

File: lib/common/client/middleware/request/remove_cookies.rb [hunk 2]
```diff
@@ -7,7 +7,8 @@ module Request
         # must be added right before http_client adapter
         class RemoveCookies < Faraday::Middleware
           def call(env)
-            @app.client.cookie_manager = nil
+            client = @app.build_connection(env)
```
Comment (ryan-mcneil) [inline lib/common/client/middleware/request/remove_cookies.rb:10]: in Faraday 1+, the client is returned via the `build_connection` method

File: lib/evss/dependents/retrieve_info_response.rb [hunk 3]
```diff
@@ -3,13 +3,13 @@
...
+    # Model for the retrieve endpoint response. Body (response_body) is passed straight through from the service.
     #
-    # @!attribute body
-    #   @return [Hash] response body
+    # @!attribute response_body
+    #   @return [Hash] response_body
     #
     class RetrieveInfoResponse < EVSS::Response
-      attribute :body, Hash
+      attribute :response_body, Hash
```
Comment (ryan-mcneil) [inline lib/evss/dependents/retrieve_info_response.rb:12]: while the `response_body` property is aliased as `body` for backwards compatibility, this doesn't seem to make the `body` property available without pulling the `response_body`

File: lib/lighthouse/benefits_claims/service.rb [hunk 4]
```diff
@@ -25,31 +25,31 @@ def get_claims(lighthouse_client_id = nil, lighthouse_rsa_key_path = nil, option
       claims
     rescue Faraday::TimeoutError
       raise BenefitsClaims::ServiceException.new({ status: 504 }), 'Lighthouse Error'
-    rescue Faraday::ClientError => e
+    rescue Faraday::ClientError, Faraday::ServerError => e
```
Comment (ryan-mcneil) [inline lib/lighthouse/benefits_claims/service.rb:28]: in Faraday 1+, 4XX-6XX errors are divided into these two categories, vs the `ClientError` that used to capture 4XX-6XX in <1

File: spec/sidekiq/evss/disability_compensation_form/submit_form526_all_claim_spec.rb [hunk 5]
```diff
@@ -423,7 +423,7 @@ def expect_retryable_error(error_class)
       context 'with an upstream service error' do
         it 'sets the transaction to "retrying"' do
           VCR.use_cassette('evss/disability_compensation_form/submit_500_with_err_msg') do
-            expect(Rails.logger).to receive(:error).twice
+            expect(Rails.logger).to receive(:error)
```
Comment (ryan-mcneil) [inline spec/sidekiq/evss/disability_compensation_form/submit_form526_all_claim_spec.rb:426]: I was instructed to adjust this expectation by codeowner @SamStuckey

**Review by aaronyoung-tw — DISMISSED** [review]
LGTM

**Review by ericboehs — APPROVED** [review]

**Pull request merged**